src/components/views/voice_messages/Clock.tsx

```diff
@@ -29,14 +29,20 @@ interface IState {
  * displayed, making it possible to see "82:29".
  */
 @replaceableComponent("views.voice_messages.Clock")
-export default class Clock extends React.PureComponent<IProps, IState> {
+export default class Clock extends React.Component<IProps, IState> {
     public constructor(props) {
         super(props);
     }
 
+    shouldComponentUpdate(nextProps: Readonly<IProps>, nextState: Readonly<IState>, nextContext: any): boolean {
+        const currentFloor = Math.floor(this.props.seconds);
+        const nextFloor = Math.floor(nextProps.seconds);
+        return currentFloor !== nextFloor;
+    }
+
     public render() {
         const minutes = Math.floor(this.props.seconds / 60).toFixed(0).padStart(2, '0');
-        const seconds = Math.round(this.props.seconds % 60).toFixed(0).padStart(2, '0'); // hide millis
+        const seconds = Math.floor(this.props.seconds % 60).toFixed(0).padStart(2, '0'); // hide millis
```

Comment:
@mvgorcum feels mathematically offended and says

> 2.999 seconds shouldn't round to 2 seconds, it should round to 3
>
> More constructively: round first, calculate minutes and seconds later

GitHub doesn't let me suggest over two lines where one was modified, but I think it's supposed to be something like this?

```js
const roundedSeconds = Math.round(this.props.seconds);
const minutes = Math.floor(roundedSeconds / 60).toFixed(0).padStart(2, '0');
const seconds = (roundedSeconds % 60).toFixed(0).padStart(2, '0'); // hide millis
```

Reply:
I'm not sure the seconds should be rounded, this is as far as I understand used during a recording. All values above 0.5 of a second would be ceiled, meaning that at 0.5 seconds of recording the clock would jump to displaying "1 second" which will skew the time perception of the user and be off by 50% at that point